scripts/test/Muon/results_tab_widget/results_tab_model_test.py

```diff
@@ -218,59 +171,48 @@ def test_model_creates_fit_selection_given_no_existing_state(self):
         _, model = create_test_model(('ws1', 'ws2'), 'func1', self.parameters,
                                      self.logs)
 
-        orig_list_state = {'ws1': [0, False, True]}
-        expected_list_state = {'ws1': [0, False, True], 'ws2': [1, True, True]}
+        orig_list_state = {'ws1_Parameters': [0, False, True]}
+        expected_list_state = {
+            'ws1_Parameters': [0, False, True],
+            'ws2_Parameters': [1, True, True]
+        }
         self.assertEqual(expected_list_state,
                          model.fit_selection(orig_list_state))
 
     def test_model_returns_no_log_selection_if_no_fits_present(self):
         model = ResultsTabModel(FittingContext())
         self.assertEqual(0, len(model.log_selection({})))
 
-    def test_model_returns_log_selection_of_first_workspace(self):
-        _, model = create_test_model(('ws1', 'ws2'), 'func1', self.parameters)
-        with mock.patch(LOG_NAMES_FUNC) as mock_log_names:
-            ws1_logs = ('run_number', 'run_start')
-            ws2_logs = ('temp', 'magnetic_field')
-
-            def side_effect(name):
-                return ws1_logs if name == 'ws1' else ws2_logs
-
-            mock_log_names.side_effect = side_effect
-            expected_selection = {
-                'run_number': [0, False, True],
-                'run_start': [1, False, True],
-            }
-
-            self.assertEqual(expected_selection, model.log_selection({}))
-
     def test_model_combines_existing_log_selection(self):
         _, model = create_test_model(('ws1', ), 'func1', self.parameters)
-        with mock.patch(LOG_NAMES_FUNC) as mock_log_names:
-            mock_log_names.return_value = [
-                'run_number', 'run_start', 'magnetic_field'
-            ]
-
-            existing_selection = {
-                'run_number': [0, False, True],
-                'run_start': [1, True, True],
-            }
-            expected_selection = deepcopy(existing_selection)
-            expected_selection.update({
-                'run_number': [0, False, True],
-                'run_start': [1, True, True],
-                'magnetic_field': [2, False, True],
-            })
-
-            self.assertDictEqual(expected_selection,
-                                 model.log_selection(existing_selection))
+        model._fit_context.log_names = mock.MagicMock()
+        model._fit_context.log_names.return_value = [
+            'run_number', 'run_start', 'magnetic_field'
+        ]
+
+        existing_selection = {
+            'run_number': [0, False, True],
+            'run_start': [1, True, True],
+        }
+        expected_selection = deepcopy(existing_selection)
+        expected_selection.update({
+            'run_number': [0, False, True],
+            'run_start': [1, True, True],
+            'magnetic_field': [2, False, True],
+        })
+
+        self.assertDictEqual(expected_selection,
```

Comment:
sorry, but being a bit slow. Why does the magnetic_field not matter in the expected_selection?

Reply:
Ah yes the model for `ListSelectorPresenter` is a dictionary of `{'name': [position, checked, enabled]}`. This is basically saying that as `magnetic_field` is a newly added log that it is in the list but not checked by default.

Reply:
I assumed we wouldn't automatically want all of the logs checked?

Reply:
ah ok, that makes sense